File: WriterApp/src/WriterApp.Web/ViewModel/Writer/WriterEditModel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace WriterApp.Web.ViewModel
{
    public class WriterEditModel
    {
        public int Id { get; set; }

        [Required]
        [StringLength(256)]
        public string FirstName { get; set; }

        [Required]
        [StringLength(256)]
        public string LastName { get; set; }

        [Required]
        public DateTime DateOfBirth { get; set; }

        [StringLength(int.MaxValue)]
        public string Biography { get; set; }
    }
}

File: WriterApp/src/WriterApp.Web/SPA/app/model/writer-edit.model.ts
export class WriterEditModel {
    constructor(
        public firstName: string,
        public lastName: string,
        public dateOfBirth: Date,
        public biography: string
    ) { }

    static fromJSON(object: any): WriterEditModel {
        return new WriterEditModel(
            object['firstName'],
            object['lastName'],
            object['dateOfBirth'],
            object['biography']
        );
    }
    static fromJSONArray(array: Array<Object>): WriterEditModel[] {
        return array.map(obj => WriterEditModel.fromJSON(obj));
    }
}
